drift_flutter/lib/src/connect.dart

```diff
@@ -67,10 +68,41 @@ final class DriftNativeOptions {
   /// a custom database path in another directory.
   final Future<String> Function()? databasePath;
 
+  /// The directory where the Drift database files are stored.
+  ///
+  /// If a custom [databasePath] is provided, this option must not be used.
+  ///
+  /// If no custom [databasePath] is provided and this option is not used, the
+  /// default directory is [DriftDatabaseDirectory.applicationDocumentsDirectory].
+  final DriftDatabaseDirectory? directory;
```

Comment:
Can we make this a `Future<Directory> Function()` and then let users use `directory: getApplicationSupportDirectory` instead?

I think this has similar ergonomics while also allowing different directory sources.

Reply:
Smart 
Let's make it a `FutureOr<Directory> Function()` and we can just keep a single argument.